### PR TITLE
Add new CI images (Focal, Bullseye)

### PR DESCRIPTION
**Pull request by janisozaur**
This adds new images for use in CI, as preparation towards C++20
support (https://github.com/OpenRCT2/OpenRCT2/issues/15501).

### CODE REVIEW

**Comment by janisozaur** [comment]
The plan is to have those new images already enabled in preparation for C++20 support and as I progress with https://github.com/OpenRCT2/OpenRCT2/pull/16107 and openrct2.org gets updated for multi-distro uploads eventually switch to the new images exclusively

**Comment by janisozaur** [comment]
As a reminder: those new builds have ccache enabled as well, initial build might take long time, but subsequent will be significantly faster.

**Comment by janisozaur** [comment]
As there was a name change in CI jobs, this PR now waits for jobs that don't exist. Actual CI run did finish successfully: https://github.com/OpenRCT2/OpenRCT2/actions/runs/1660664102

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -220,10 +220,22 @@ jobs:
...
             build_flags: -DCMAKE_POSITION_INDEPENDENT_CODE=on -DCMAKE_CXX_FLAGS="-g -gz"
+          - platform: x86_64
+            distro: focal
+            image: openrct2/openrct2-build:5-focal
+            cache_key: linux-portable
+            build_flags: -DCMAKE_POSITION_INDEPENDENT_CODE=on -DCMAKE_CXX_FLAGS="-g -gz"
+          - platform: x86_64
+            distro: bullseye
+            image: openrct2/openrct2-build:5-bullseye
+            cache_key: linux-portable
```
Comment (tupaschoal) [inline .github/workflows/ci.yml:235]: Should the keys have been different?

Comment (janisozaur) [reply .github/workflows/ci.yml:235]: Yup, thanks

Comment (janisozaur) [reply .github/workflows/ci.yml:235]: I will send an update later today

**Pull request merged**